### ISSUE TITLE
Detecting Mate isn't working properly

### ISSUE THREAD

**Issue by jcgruenhage** [jcgruenhage/Chess#5]
The King can run into situation where he can be captured:

``` java
King from H6 to H5; Turn 268
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    |    |    |    |    |    |    |    |
  -----------------------------------------
6 |    | Pw |    | Pw |    |    |    |    |
  -----------------------------------------
5 |    |    |    |    | Nw |    |    | Kb |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    | Pb |
  -----------------------------------------
2 |    |    |    |    |    |    |    |    |
  -----------------------------------------
1 |    |    |    |    |    |    | Kw |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

King from F2 to G1; Turn 269
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    |    |    |    |    |    |    |    |
  -----------------------------------------
6 |    | Pw |    | Pw |    |    |    | Kb |
  -----------------------------------------
5 |    |    |    |    | Nw |    |    |    |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    | Pb |
  -----------------------------------------
2 |    |    |    |    |    |    |    |    |
  -----------------------------------------
1 |    |    |    |    |    |    | Kw |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

King from H5 to H6; Turn 270
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    |    |    |    |    |    |    |    |
  -----------------------------------------
6 |    | Pw |    | Pw |    |    |    | Kb |
  -----------------------------------------
5 |    |    |    |    | Nw |    |    |    |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    | Pb |
  -----------------------------------------
2 |    |    |    |    |    |    | Kw |    |
  -----------------------------------------
1 |    |    |    |    |    |    |    |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

King from G1 to G2; Turn 271
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    |    |    |    |    |    |    |    |
  -----------------------------------------
6 |    | Pw |    | Pw |    |    |    | Kb |
  -----------------------------------------
5 |    |    |    |    | Nw |    |    |    |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    |    |
  -----------------------------------------
2 |    |    |    |    |    |    | Pb |    |
  -----------------------------------------
1 |    |    |    |    |    |    |    |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

Pawn from H3 to G2; Turn 272
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    |    |    |    |    |    |    |    |
  -----------------------------------------
6 |    | Pw |    | Pw |    |    | Nw | Kb |
  -----------------------------------------
5 |    |    |    |    |    |    |    |    |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    |    |
  -----------------------------------------
2 |    |    |    |    |    |    | Pb |    |
  -----------------------------------------
1 |    |    |    |    |    |    |    |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

Knight from E5 to G6; Turn 273
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    |    |    |    |    |    |    |    |
  -----------------------------------------
6 |    | Pw |    | Pw |    |    | Kb |    |
  -----------------------------------------
5 |    |    |    |    |    |    |    |    |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    |    |
  -----------------------------------------
2 |    |    |    |    |    |    | Pb |    |
  -----------------------------------------
1 |    |    |    |    |    |    |    |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

King from H6 to G6; Turn 274
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    | Pw |    |    |    |    |    |    |
  -----------------------------------------
6 |    |    |    | Pw |    |    | Kb |    |
  -----------------------------------------
5 |    |    |    |    |    |    |    |    |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    |    |
  -----------------------------------------
2 |    |    |    |    |    |    | Pb |    |
  -----------------------------------------
1 |    |    |    |    |    |    |    |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

Pawn from B6 to B7; Turn 275
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    | Pw |    |    |    |    |    |    |
  -----------------------------------------
6 |    |    |    | Pw |    |    | Kb |    |
  -----------------------------------------
5 |    |    |    |    |    |    |    |    |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    |    |
  -----------------------------------------
2 |    |    |    |    |    |    |    |    |
  -----------------------------------------
1 |    |    |    |    |    |    | Pb |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

Pawn from G2 to G1; Turn 276
  -----------------------------------------
8 |    |    |    |    |    |    |    |    |
  -----------------------------------------
7 |    | Pw |    | Pw |    |    |    |    |
  -----------------------------------------
6 |    |    |    |    |    |    | Kb |    |
  -----------------------------------------
5 |    |    |    |    |    |    |    |    |
  -----------------------------------------
4 | Pw |    |    |    |    |    |    |    |
  -----------------------------------------
3 |    |    |    |    |    |    |    |    |
  -----------------------------------------
2 |    |    |    |    |    |    |    |    |
  -----------------------------------------
1 |    |    |    |    |    |    | Pb |    |
  -----------------------------------------
     A    B    C    D    E    F    G    H  

Pawn from D6 to D7; Turn 277
Exception in thread "main" java.lang.IllegalArgumentException
    at edu.gymneureut.informatik.rattenschach.model.Field.<init>(Field.java:16)
    at edu.gymneureut.informatik.rattenschach.model.figures.Pawn.getPossibleMoves(Pawn.java:63)
    at edu.gymneureut.informatik.rattenschach.model.Player.move(Player.java:160)
    at edu.gymneureut.informatik.rattenschach.model.Game.act(Game.java:120)
    at edu.gymneureut.informatik.rattenschach.model.Game.play(Game.java:101)
    at edu.gymneureut.informatik.rattenschach.control.Main.main(Main.java:60)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)

Process finished with exit code 1
```

Also, Pawn Promotion Bug is here as well.

**Comment by jcgruenhage** [comment]
Now in the JGGGUI and with excption due to hotfix KingCapture

![jgamegrid v2 46 _004](https://cloud.githubusercontent.com/assets/6861958/14912638/d89657dc-0dfd-11e6-8979-3b5e513ee2c9.png)

![jgamegrid fatal error_003](https://cloud.githubusercontent.com/assets/6861958/14912645/dd587462-0dfd-11e6-9ffc-cdd0b6deccd6.png)

**Comment by jcgruenhage** [comment]
Now that it's throwing an exception, I notice it much more frequently, nearly every game ends in this way.

**Comment by jcgruenhage** [comment]
The Bug is still present, but i can't figure out why. if there is still time before the deadline, maybe i'll look into it again.

**Comment by jcgruenhage** [comment]
Fixed in latest commit 134c983eed6f7a631c1f53a34468cca0507a1cff
